From 4ba0e86d3d3c94584bae555ce3de782d4facb794 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 18:43:08 -0500
Subject: [PATCH 01/10] Death Knight Skip Module

Skip Death Knight Module
Original Module From Single Player Project Consolidated Skip Module
Rewritten for TC 335 By Single Player Project Developer MDic
Original Concept from conanhun513
Works for both horde and alliance new characters.
When Module is active (Conf = 1) and then a new character Death Knight is created, intro is played (not skippable). Character is then given flags and triggers based on quest completions for the starter areas and give the initial turn in quest after leaving the ebon hold. Alliance DK will be in storm wind with the quest https://www.wowhead.com/quest=13188/where-kings-walk ready for turn in, and Horde DK will be in Orgrimmar with the quest https://www.wowhead.com/quest=13189/saurfangs-blessing. All obtainable key Character Items are in the player's inventory or auto equipped, (Tabard, Weapons, Armors, and such). Spells such as Death Gate works and does return you to the appropriate ebon hold after the DK start zone chain quests are finished.
Runeforging, journeyman riding skipp and deathcharge mount is learned as well.
Tested and working on current 335 with all items and quests and triggers researched for 335 branch.

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp          | 86 +++++++++++++++++++
 .../scripts/Custom/custom_script_loader.cpp   |  3 +
 src/server/worldserver/worldserver.conf.dist  | 17 ++++
 3 files changed, 106 insertions(+)
 create mode 100644 src/server/scripts/Custom/SkipDK.cpp

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
new file mode 100644
index 0000000000..7f288503a4
--- /dev/null
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -0,0 +1,86 @@
+/* Skip Death Knight Module
+   Original Module From Single Player Project Consolidated Skip Module
+   Rewritten for TC 335 By Single Player Project Developer MDic
+   Original Concept from conanhun513
+*/
+
+#include "ScriptMgr.h"
+#include "Player.h"
+#include "Config.h"
+#include "Chat.h"
+#include "ObjectAccessor.h"
+#include "ObjectMgr.h"
+#include "SharedDefines.h"
+#include "World.h"
+
+class Trinitycore_skip_deathknight : public PlayerScript
+{
+public:
+    Trinitycore_skip_deathknight() : PlayerScript("Trinitycore_skip_deathknight") { }
+
+    void OnLogin(Player* Player, bool firstLogin) override
+    {
+        if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Enable", true))
+        {
+            if (Player->GetAreaId() == 4342)
+            {
+                if (!firstLogin)
+                    return;
+                Player->SetLevel(58);
+                Player->LearnSpell(53428, false);//runeforging
+                Player->LearnSpell(53441, false);//runeforging
+                Player->LearnSpell(53344, false);//runeforging
+                Player->LearnSpell(62158, false);//runeforging
+                Player->LearnSpell(33391, false);//journeyman riding
+                Player->LearnSpell(54586, false);//runeforging credit
+                Player->LearnSpell(48778, false);//acherus deathcharger
+                Player->LearnSkillRewardedSpells(776, 375);//Runeforging
+                Player->LearnSkillRewardedSpells(960, 375);//Runeforging
+                Player->EquipNewItem(EQUIPMENT_SLOT_HEAD, 38661, true);//Greathelm of the Scourge Champion
+                Player->EquipNewItem(EQUIPMENT_SLOT_WRISTS, 38666, true);//Plated Saronite Bracers
+                Player->EquipNewItem(EQUIPMENT_SLOT_WAIST, 38668, true);//The Plaguebringer's Girdle
+                Player->EquipNewItem(EQUIPMENT_SLOT_HANDS, 38667, true);//Bloodbane's Gauntlets of Command
+                Player->EquipNewItem(EQUIPMENT_SLOT_CHEST, 38665, true);//Saronite War Plate
+                Player->EquipNewItem(EQUIPMENT_SLOT_LEGS, 38669, true);//Engraved Saronite Legplates
+                Player->EquipNewItem(EQUIPMENT_SLOT_SHOULDERS, 38663, true);// Blood-Soaked Saronite Plated Spaulders
+                Player->EquipNewItem(EQUIPMENT_SLOT_FEET, 38670, true);//Greaves of the Slaughter
+                Player->EquipNewItem(EQUIPMENT_SLOT_TRINKET1, 38675, true);//Signet of the Dark Brotherhood
+                Player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
+                Player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
+                Player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
+                Player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
+                Player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
+                Player->AddItem(39322, true);//Sky Darkener's Shroud of the Unholy
+                Player->AddItem(38632, true);//Greatsword of the Ebon Blade
+                Player->AddItem(6948, true);//Hearthstone
+                Player->AddItem(38707, true);//Runed Soulblade
+                Player->AddItem(40483, true);//Insignia of the Scourge
+                                
+                if (Player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
+                {
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), NULL);
+                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), NULL, Player);
+                }
+                if (Player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
+                {
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12801), NULL);
+                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), NULL, Player);
+                }
+                if (Player->GetTeam() == ALLIANCE && Player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13188), NULL);
+                else if (Player->GetTeam() == HORDE && Player->GetQuestStatus(13189) == QUEST_STATUS_NONE)//Saurfang's Blessing
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13189), NULL);
+                if (Player->GetTeam() == ALLIANCE)
+                    Player->TeleportTo(0, -8833.37f, 628.62f, 94.00f, 1.06f);//Stormwind
+                else
+                    Player->TeleportTo(1, 1569.59f, -4397.63f, 16.06f, 0.54f);//Orgrimmar
+                ObjectAccessor::SaveAllPlayers();//Save
+            }
+        }
+    }
+};
+
+void AddSC_skip_StarterArea()
+{
+    new Trinitycore_skip_deathknight;
+}
diff --git a/src/server/scripts/Custom/custom_script_loader.cpp b/src/server/scripts/Custom/custom_script_loader.cpp
index 9e5e9ba2bf..702789c280 100644
--- a/src/server/scripts/Custom/custom_script_loader.cpp
+++ b/src/server/scripts/Custom/custom_script_loader.cpp
@@ -19,6 +19,9 @@
 
 // The name of this function should match:
 // void Add${NameOfDirectory}Scripts()
+void AddSC_skip_StarterArea();
+
 void AddCustomScripts()
 {
+    AddSC_skip_StarterArea();
 }
diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index 0a79664476..3d5657791b 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4089,3 +4089,20 @@ Metric.OverallStatusInterval = 1
 
 #
 ###################################################################################################
+
+###################################################################################################
+#Skip Death Knight Starting Area#
+#################################
+#
+#    Skip Death Knight Starting Area
+#        Enable this if you want to skip the Deathknight starting area
+#		 Start with level 58 DK with Runeforging. With Portable Runeforge Skill
+#
+#    Skip.DeathKnight.Enable
+#        Default:    1 (enabled)
+#                    0 (disabled)
+
+Skip.Deathknight.Enable = 0
+
+#
+###################################################################################################
\ No newline at end of file

From 478c4dd3eeeaec37e8d50ba1b4fd37c44aeb303e Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 18:45:41 -0500
Subject: [PATCH 02/10] Fix white space

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/worldserver/worldserver.conf.dist | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index 3d5657791b..7b91eebe8b 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4096,7 +4096,7 @@ Metric.OverallStatusInterval = 1
 #
 #    Skip Death Knight Starting Area
 #        Enable this if you want to skip the Deathknight starting area
-#		 Start with level 58 DK with Runeforging. With Portable Runeforge Skill
+#        Start with level 58 DK with Runeforging. With Portable Runeforge Skill
 #
 #    Skip.DeathKnight.Enable
 #        Default:    1 (enabled)

From 0c20d959d49cd2c3865ee0df88343dec8f3bfd06 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 18:52:39 -0500
Subject: [PATCH 03/10] more whitespace removal.

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index 7f288503a4..f47fb6612d 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -55,7 +55,7 @@ class Trinitycore_skip_deathknight : public PlayerScript
                 Player->AddItem(6948, true);//Hearthstone
                 Player->AddItem(38707, true);//Runed Soulblade
                 Player->AddItem(40483, true);//Insignia of the Scourge
-                                
+
                 if (Player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
                 {
                     Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), NULL);

From 451e797fc78135c37c9bd848142e19cd8b76a7cd Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 19:06:11 -0500
Subject: [PATCH 04/10] adjust null to nullptr

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index f47fb6612d..022d032ea5 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -58,18 +58,18 @@ class Trinitycore_skip_deathknight : public PlayerScript
 
                 if (Player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
                 {
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), NULL);
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), nullptr);
                     Player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), NULL, Player);
                 }
                 if (Player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
                 {
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12801), NULL);
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12801), nullptr);
                     Player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), NULL, Player);
                 }
                 if (Player->GetTeam() == ALLIANCE && Player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13188), NULL);
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13188), nullptr);
                 else if (Player->GetTeam() == HORDE && Player->GetQuestStatus(13189) == QUEST_STATUS_NONE)//Saurfang's Blessing
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13189), NULL);
+                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13189), nullptr);
                 if (Player->GetTeam() == ALLIANCE)
                     Player->TeleportTo(0, -8833.37f, 628.62f, 94.00f, 1.06f);//Stormwind
                 else

From f668cca5fc5f1d56d6609f22f73c3c5ec955e82e Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 19:18:26 -0500
Subject: [PATCH 05/10] NULL to false

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index 022d032ea5..dba84557af 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -59,12 +59,12 @@ class Trinitycore_skip_deathknight : public PlayerScript
                 if (Player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
                 {
                     Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), nullptr);
-                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), NULL, Player);
+                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), false, Player);
                 }
                 if (Player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
                 {
                     Player->AddQuest(sObjectMgr->GetQuestTemplate(12801), nullptr);
-                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), NULL, Player);
+                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), false, Player);
                 }
                 if (Player->GetTeam() == ALLIANCE && Player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
                     Player->AddQuest(sObjectMgr->GetQuestTemplate(13188), nullptr);

From a7d5768ae7cd021bc3fa431469cbabd5b05d6548 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 19:30:43 -0500
Subject: [PATCH 06/10] conf correction and announcement module addon

conf correction, announcement module addon, and internal description correction

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp         | 10 +++++++++-
 src/server/worldserver/worldserver.conf.dist | 15 +++++++++++----
 2 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index dba84557af..f5b8de0e2a 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -18,6 +18,14 @@ class Trinitycore_skip_deathknight : public PlayerScript
 public:
     Trinitycore_skip_deathknight() : PlayerScript("Trinitycore_skip_deathknight") { }
 
+    void OnLogin(Player* Player, bool /*firstLogin*/) override
+    {
+        if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Announce.enable", true))
+        {
+            ChatHandler(Player->GetSession()).SendSysMessage("This server is running the |cff4CFF00Trinitycore Skip Deathknight Starter |rmodule.");
+        }
+    }
+
     void OnLogin(Player* Player, bool firstLogin) override
     {
         if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Enable", true))
@@ -50,7 +58,7 @@ class Trinitycore_skip_deathknight : public PlayerScript
                 Player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
                 Player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
                 Player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
-                Player->AddItem(39322, true);//Sky Darkener's Shroud of the Unholy
+                Player->AddItem(39322, true);//Shroud of the North Wind
                 Player->AddItem(38632, true);//Greatsword of the Ebon Blade
                 Player->AddItem(6948, true);//Hearthstone
                 Player->AddItem(38707, true);//Runed Soulblade
diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index 7b91eebe8b..cd787db68d 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4098,11 +4098,18 @@ Metric.OverallStatusInterval = 1
 #        Enable this if you want to skip the Deathknight starting area
 #        Start with level 58 DK with Runeforging. With Portable Runeforge Skill
 #
-#    Skip.DeathKnight.Enable
-#        Default:    1 (enabled)
-#                    0 (disabled)
+#    Skip.Deathknight.Starter.Announce.enable
+#         Description: Announces Module when player logs in.
+#         Default:    0 - (Disabled)
+#                     1 - (Enabled)
+
+Skip.Deathknight.Starter.Announce.enable = 0
+
+#    Skip.DeathKnight.Starter.Enable
+#         Default:    0 - (Disabled)
+#                     1 - (Enabled)
 
-Skip.Deathknight.Enable = 0
+Skip.Deathknight.Starter.Enable = 0
 
 #
 ###################################################################################################
\ No newline at end of file

From 6e1f28c31d2aaa725f2b3cf59ad94489b7822b32 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Tue, 2 Feb 2021 19:33:52 -0500
Subject: [PATCH 07/10] shift and correction

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index f5b8de0e2a..ad86f57e65 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -18,16 +18,13 @@ class Trinitycore_skip_deathknight : public PlayerScript
 public:
     Trinitycore_skip_deathknight() : PlayerScript("Trinitycore_skip_deathknight") { }
 
-    void OnLogin(Player* Player, bool /*firstLogin*/) override
+    void OnLogin(Player* Player, bool firstLogin) override
     {
         if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Announce.enable", true))
         {
             ChatHandler(Player->GetSession()).SendSysMessage("This server is running the |cff4CFF00Trinitycore Skip Deathknight Starter |rmodule.");
         }
-    }
 
-    void OnLogin(Player* Player, bool firstLogin) override
-    {
         if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Enable", true))
         {
             if (Player->GetAreaId() == 4342)

From 1df63e522003c1fa28ba349a2f14d2fbe620fb88 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Wed, 3 Feb 2021 08:55:05 -0500
Subject: [PATCH 08/10] Announcement Shift, GM DK Conf and DK start level
 configurable

Shifted Announcement Module to its own void.
Added Death Knight Configurable Start Level via conf
Added option for module to be active for GM accounts only.

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/scripts/Custom/SkipDK.cpp         | 166 ++++++++++++++-----
 src/server/worldserver/worldserver.conf.dist |  21 ++-
 2 files changed, 139 insertions(+), 48 deletions(-)

diff --git a/src/server/scripts/Custom/SkipDK.cpp b/src/server/scripts/Custom/SkipDK.cpp
index ad86f57e65..4a0c250b5b 100644
--- a/src/server/scripts/Custom/SkipDK.cpp
+++ b/src/server/scripts/Custom/SkipDK.cpp
@@ -4,81 +4,154 @@
    Original Concept from conanhun513
 */
 
+#include "AccountMgr.h"
 #include "ScriptMgr.h"
 #include "Player.h"
 #include "Config.h"
+#include "Common.h"
 #include "Chat.h"
 #include "ObjectAccessor.h"
 #include "ObjectMgr.h"
+#include "Player.h"
 #include "SharedDefines.h"
 #include "World.h"
+#include "WorldSession.h"
 
-class Trinitycore_skip_deathknight : public PlayerScript
+class Trinitycore_skip_deathknight_announce : public PlayerScript
 {
 public:
-    Trinitycore_skip_deathknight() : PlayerScript("Trinitycore_skip_deathknight") { }
+    Trinitycore_skip_deathknight_announce() : PlayerScript("Trinitycore_skip_deathknight_announce") { }
 
-    void OnLogin(Player* Player, bool firstLogin) override
+    void OnLogin(Player* Player, bool /*firstLogin*/) override
     {
         if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Announce.enable", true))
         {
             ChatHandler(Player->GetSession()).SendSysMessage("This server is running the |cff4CFF00Trinitycore Skip Deathknight Starter |rmodule.");
         }
+    }
+};
+
+class Trinitycore_skip_deathknight : public PlayerScript
+{
+public:
+    Trinitycore_skip_deathknight() : PlayerScript("Trinitycore_skip_deathknight") { }
+
+    void OnLogin(Player* player, bool firstLogin) override
+    {
+        int DKL = sConfigMgr->GetFloatDefault("Skip.Deathknight.Start.Level", 58);
 
         if (sConfigMgr->GetBoolDefault("Skip.Deathknight.Starter.Enable", true))
         {
-            if (Player->GetAreaId() == 4342)
+            if (player->GetAreaId() == 4342)
+            {
+                if (!firstLogin)
+                    return;
+                player->SetLevel(DKL);
+                player->LearnSpell(53428, false);//runeforging
+                player->LearnSpell(53441, false);//runeforging
+                player->LearnSpell(53344, false);//runeforging
+                player->LearnSpell(62158, false);//runeforging
+                player->LearnSpell(33391, false);//journeyman riding
+                player->LearnSpell(54586, false);//runeforging credit
+                player->LearnSpell(48778, false);//acherus deathcharger
+                player->LearnSkillRewardedSpells(776, 375);//Runeforging
+                player->LearnSkillRewardedSpells(960, 375);//Runeforging
+                player->EquipNewItem(EQUIPMENT_SLOT_HEAD, 38661, true);//Greathelm of the Scourge Champion
+                player->EquipNewItem(EQUIPMENT_SLOT_WRISTS, 38666, true);//Plated Saronite Bracers
+                player->EquipNewItem(EQUIPMENT_SLOT_WAIST, 38668, true);//The Plaguebringer's Girdle
+                player->EquipNewItem(EQUIPMENT_SLOT_HANDS, 38667, true);//Bloodbane's Gauntlets of Command
+                player->EquipNewItem(EQUIPMENT_SLOT_CHEST, 38665, true);//Saronite War Plate
+                player->EquipNewItem(EQUIPMENT_SLOT_LEGS, 38669, true);//Engraved Saronite Legplates
+                player->EquipNewItem(EQUIPMENT_SLOT_SHOULDERS, 38663, true);// Blood-Soaked Saronite Plated Spaulders
+                player->EquipNewItem(EQUIPMENT_SLOT_FEET, 38670, true);//Greaves of the Slaughter
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET1, 38675, true);//Signet of the Dark Brotherhood
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
+                player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
+                player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
+                player->AddItem(39322, true);//Shroud of the North Wind
+                player->AddItem(38632, true);//Greatsword of the Ebon Blade
+                player->AddItem(6948, true);//Hearthstone
+                player->AddItem(38707, true);//Runed Soulblade
+                player->AddItem(40483, true);//Insignia of the Scourge
+
+                if (player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
+                {
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(12657), nullptr);
+                    player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), false, player);
+                }
+                if (player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
+                {
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(12801), nullptr);
+                    player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), false, player);
+                }
+                if (player->GetTeam() == ALLIANCE && player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(13188), nullptr);
+                else if (player->GetTeam() == HORDE && player->GetQuestStatus(13189) == QUEST_STATUS_NONE)//Saurfang's Blessing
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(13189), nullptr);
+                if (player->GetTeam() == ALLIANCE)
+                    player->TeleportTo(0, -8833.37f, 628.62f, 94.00f, 1.06f);//Stormwind
+                else
+                    player->TeleportTo(1, 1569.59f, -4397.63f, 16.06f, 0.54f);//Orgrimmar
+                ObjectAccessor::SaveAllPlayers();//Save
+            }
+        }
+
+        if (sConfigMgr->GetBoolDefault("GM.Skip.Deathknight.Starter.Enable", true))
+        {
+            if (player->GetSession()->GetSecurity() >= SEC_MODERATOR && player->GetAreaId() == 4342)
             {
                 if (!firstLogin)
                     return;
-                Player->SetLevel(58);
-                Player->LearnSpell(53428, false);//runeforging
-                Player->LearnSpell(53441, false);//runeforging
-                Player->LearnSpell(53344, false);//runeforging
-                Player->LearnSpell(62158, false);//runeforging
-                Player->LearnSpell(33391, false);//journeyman riding
-                Player->LearnSpell(54586, false);//runeforging credit
-                Player->LearnSpell(48778, false);//acherus deathcharger
-                Player->LearnSkillRewardedSpells(776, 375);//Runeforging
-                Player->LearnSkillRewardedSpells(960, 375);//Runeforging
-                Player->EquipNewItem(EQUIPMENT_SLOT_HEAD, 38661, true);//Greathelm of the Scourge Champion
-                Player->EquipNewItem(EQUIPMENT_SLOT_WRISTS, 38666, true);//Plated Saronite Bracers
-                Player->EquipNewItem(EQUIPMENT_SLOT_WAIST, 38668, true);//The Plaguebringer's Girdle
-                Player->EquipNewItem(EQUIPMENT_SLOT_HANDS, 38667, true);//Bloodbane's Gauntlets of Command
-                Player->EquipNewItem(EQUIPMENT_SLOT_CHEST, 38665, true);//Saronite War Plate
-                Player->EquipNewItem(EQUIPMENT_SLOT_LEGS, 38669, true);//Engraved Saronite Legplates
-                Player->EquipNewItem(EQUIPMENT_SLOT_SHOULDERS, 38663, true);// Blood-Soaked Saronite Plated Spaulders
-                Player->EquipNewItem(EQUIPMENT_SLOT_FEET, 38670, true);//Greaves of the Slaughter
-                Player->EquipNewItem(EQUIPMENT_SLOT_TRINKET1, 38675, true);//Signet of the Dark Brotherhood
-                Player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
-                Player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
-                Player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
-                Player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
-                Player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
-                Player->AddItem(39322, true);//Shroud of the North Wind
-                Player->AddItem(38632, true);//Greatsword of the Ebon Blade
-                Player->AddItem(6948, true);//Hearthstone
-                Player->AddItem(38707, true);//Runed Soulblade
-                Player->AddItem(40483, true);//Insignia of the Scourge
+                player->SetLevel(DKL);
+                player->LearnSpell(53428, false);//runeforging
+                player->LearnSpell(53441, false);//runeforging
+                player->LearnSpell(53344, false);//runeforging
+                player->LearnSpell(62158, false);//runeforging
+                player->LearnSpell(33391, false);//journeyman riding
+                player->LearnSpell(54586, false);//runeforging credit
+                player->LearnSpell(48778, false);//acherus deathcharger
+                player->LearnSkillRewardedSpells(776, 375);//Runeforging
+                player->LearnSkillRewardedSpells(960, 375);//Runeforging
+                player->EquipNewItem(EQUIPMENT_SLOT_HEAD, 38661, true);//Greathelm of the Scourge Champion
+                player->EquipNewItem(EQUIPMENT_SLOT_WRISTS, 38666, true);//Plated Saronite Bracers
+                player->EquipNewItem(EQUIPMENT_SLOT_WAIST, 38668, true);//The Plaguebringer's Girdle
+                player->EquipNewItem(EQUIPMENT_SLOT_HANDS, 38667, true);//Bloodbane's Gauntlets of Command
+                player->EquipNewItem(EQUIPMENT_SLOT_CHEST, 38665, true);//Saronite War Plate
+                player->EquipNewItem(EQUIPMENT_SLOT_LEGS, 38669, true);//Engraved Saronite Legplates
+                player->EquipNewItem(EQUIPMENT_SLOT_SHOULDERS, 38663, true);// Blood-Soaked Saronite Plated Spaulders
+                player->EquipNewItem(EQUIPMENT_SLOT_FEET, 38670, true);//Greaves of the Slaughter
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET1, 38675, true);//Signet of the Dark Brotherhood
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
+                player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
+                player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
+                player->AddItem(39322, true);//Shroud of the North Wind
+                player->AddItem(38632, true);//Greatsword of the Ebon Blade
+                player->AddItem(6948, true);//Hearthstone
+                player->AddItem(38707, true);//Runed Soulblade
+                player->AddItem(40483, true);//Insignia of the Scourge
 
-                if (Player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
+                if (player->GetQuestStatus(12657) == QUEST_STATUS_NONE)//The Might Of The Scourge
                 {
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12657), nullptr);
-                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), false, Player);
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(12657), nullptr);
+                    player->RewardQuest(sObjectMgr->GetQuestTemplate(12657), false, player);
                 }
-                if (Player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
+                if (player->GetQuestStatus(12801) == QUEST_STATUS_NONE)//The Light of Dawn
                 {
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(12801), nullptr);
-                    Player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), false, Player);
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(12801), nullptr);
+                    player->RewardQuest(sObjectMgr->GetQuestTemplate(12801), false, player);
                 }
-                if (Player->GetTeam() == ALLIANCE && Player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13188), nullptr);
-                else if (Player->GetTeam() == HORDE && Player->GetQuestStatus(13189) == QUEST_STATUS_NONE)//Saurfang's Blessing
-                    Player->AddQuest(sObjectMgr->GetQuestTemplate(13189), nullptr);
-                if (Player->GetTeam() == ALLIANCE)
-                    Player->TeleportTo(0, -8833.37f, 628.62f, 94.00f, 1.06f);//Stormwind
+                if (player->GetTeam() == ALLIANCE && player->GetQuestStatus(13188) == QUEST_STATUS_NONE)//Where Kings Walk
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(13188), nullptr);
+                else if (player->GetTeam() == HORDE && player->GetQuestStatus(13189) == QUEST_STATUS_NONE)//Saurfang's Blessing
+                    player->AddQuest(sObjectMgr->GetQuestTemplate(13189), nullptr);
+                if (player->GetTeam() == ALLIANCE)
+                    player->TeleportTo(0, -8833.37f, 628.62f, 94.00f, 1.06f);//Stormwind
                 else
-                    Player->TeleportTo(1, 1569.59f, -4397.63f, 16.06f, 0.54f);//Orgrimmar
+                    player->TeleportTo(1, 1569.59f, -4397.63f, 16.06f, 0.54f);//Orgrimmar
                 ObjectAccessor::SaveAllPlayers();//Save
             }
         }
@@ -87,5 +160,6 @@ class Trinitycore_skip_deathknight : public PlayerScript
 
 void AddSC_skip_StarterArea()
 {
+    new Trinitycore_skip_deathknight_announce;
     new Trinitycore_skip_deathknight;
 }
diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index cd787db68d..12bc25cc17 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4090,6 +4090,7 @@ Metric.OverallStatusInterval = 1
 #
 ###################################################################################################
 
+
 ###################################################################################################
 #Skip Death Knight Starting Area#
 #################################
@@ -4103,13 +4104,29 @@ Metric.OverallStatusInterval = 1
 #         Default:    0 - (Disabled)
 #                     1 - (Enabled)
 
-Skip.Deathknight.Starter.Announce.enable = 0
+Skip.Deathknight.Starter.Announce.enable = 1
 
 #    Skip.DeathKnight.Starter.Enable
+#         Active Module for all new accounts regardless of account level.
+#         Default:    0 - (Disabled)
+#                     1 - (Enabled)
+
+Skip.Deathknight.Starter.Enable = 1
+
+#    GM.Skip.Deathknight.Starter.Enable
+#         Active Module for all new accounts of a GM account level ONLY.
+#         Ensure you have Skip.Deathknight.Starter.Enable set to 0 if using this conf.
 #         Default:    0 - (Disabled)
 #                     1 - (Enabled)
 
-Skip.Deathknight.Starter.Enable = 0
+GM.Skip.Deathknight.Starter.Enable = 0
+
+#    Skip.Deathknight.Start.Level
+#         Adjusted Death Knight Start Level if module is used.
+#         Default:    58 - (Default)
+#
+
+Skip.Deathknight.Start.Level = 58
 
 #
 ###################################################################################################
\ No newline at end of file

From cbb62f9dc40a6de8763d8d19fbe167c2184f8812 Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Wed, 3 Feb 2021 08:56:35 -0500
Subject: [PATCH 09/10] blank line removal

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/worldserver/worldserver.conf.dist | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index 12bc25cc17..564413c3fd 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4090,7 +4090,6 @@ Metric.OverallStatusInterval = 1
 #
 ###################################################################################################
 
-
 ###################################################################################################
 #Skip Death Knight Starting Area#
 #################################

From d92cfc0dfb25e3114d428bf4f14de831520a6a6e Mon Sep 17 00:00:00 2001
From: acidmanifesto <joshua.lee.betts@gmail.com>
Date: Wed, 3 Feb 2021 17:30:24 -0500
Subject: [PATCH 10/10] Corrected Description.

Portable Rune Forge was a Legion V2 feature that is not possible on 334.

Co-Authored-By: ConanHUN <conan_513@hotmail.com>
Co-Authored-By: Serithul <62258982+serithul@users.noreply.github.com>
Co-Authored-By: Warlord Zul'Jin <42502821+takasi689@users.noreply.github.com>
---
 src/server/worldserver/worldserver.conf.dist | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/worldserver/worldserver.conf.dist b/src/server/worldserver/worldserver.conf.dist
index 564413c3fd..24728797c5 100644
--- a/src/server/worldserver/worldserver.conf.dist
+++ b/src/server/worldserver/worldserver.conf.dist
@@ -4096,7 +4096,7 @@ Metric.OverallStatusInterval = 1
 #
 #    Skip Death Knight Starting Area
 #        Enable this if you want to skip the Deathknight starting area
-#        Start with level 58 DK with Runeforging. With Portable Runeforge Skill
+#        Start with level 58 DK with Runeforging.
 #
 #    Skip.Deathknight.Starter.Announce.enable
 #         Description: Announces Module when player logs in.